inlong-audit/audit-proxy/src/main/java/org/apache/inlong/audit/sink/KafkaSink.java

```diff
@@ -0,0 +1,430 @@
+                    totalKafkaSuccSendSize.set(0);
+                    totalKafkaSuccSendCnt.set(0);
+                }
+
+            } catch (Exception e) {
+                logger.info("tubePerformanceTask error", e);
+            }
+        }
+    }
+
+    public void handleMessageSendSuccess(EventStat es) {
+        // Statistics tube performance
+        totalKafkaSuccSendCnt.incrementAndGet();
+        totalKafkaSuccSendSize.addAndGet(es.getEvent().getBody().length);
+
+        // add to sinkCounter
+        sinkCounter.incrementEventDrainSuccessCount();
+        currentSuccessSendCnt.incrementAndGet();
+        long nowCnt = currentSuccessSendCnt.get();
+        long oldCnt = lastSuccessSendCnt.get();
+        if (nowCnt % logEveryNEvents == 0 && nowCnt != lastSuccessSendCnt.get()) {
+            lastSuccessSendCnt.set(nowCnt);
+            t2 = System.currentTimeMillis();
+            logger.info("KafkaSink {}, succ put {} events to kafka, in the past {} millisecond",
+                    getName(), (nowCnt - oldCnt), (t2 - t1));
+            t1 = t2;
+        }
+    }
+
+    private void resendEvent(EventStat es, boolean isDecrement) {
+        try {
+            if (es == null || es.getEvent() == null) {
+                return;
+            }
+            if (!resendQueue.offer(es)) {
+                FailoverChannelProcessorHolder.getChannelProcessor().processEvent(es.getEvent());
+            }
+        } catch (Throwable throwable) {
+            logger.error("resendEvent e = {}", throwable);
+        }
+    }
+
+    class SinkTask implements Runnable {
+
+        @Override
+        public void run() {
+            logger.info("Sink task {} started.", Thread.currentThread().getName());
+            while (canSend) {
+                boolean decrementFlag = false;
+                Event event = null;
+                EventStat eventStat = null;
+                try {
+                    if (KafkaSink.this.overflow) {
+                        KafkaSink.this.overflow = false;
+                        Thread.sleep(10);
+                    }
+                    if (!resendQueue.isEmpty()) {
+                        // Send the data in the retry queue first
+                        eventStat = resendQueue.poll();
+                        if (eventStat != null) {
+                            event = eventStat.getEvent();
+                        }
+                    } else {
+                        event = eventQueue.take();
+                        eventStat = new EventStat(event);
+                        sinkCounter.incrementEventDrainAttemptCount();
+                    }
+
+                    if (event == null) {
+                        continue;
+                    }
+
+                    if (topic == null || topic.equals("")) {
+                        logger.warn("no topic specified in event header, just skip this event");
+                        continue;
+                    }
```

Comment:
1. Merge this two checks into one;
2. Use `StringUtils.isBlank()` to check the topic is null or not.

Reply:
Done.